Source/ARTAuth.m

```diff
@@ -548,30 +548,43 @@ - (void)authorize:(ARTTokenParams *)tokenParams options:(ARTAuthOptions *)authOp
     __weak id<ARTAuthDelegate> lastDelegate = self.delegate;
 
     NSString *authorizeId = [[NSUUID new] UUIDString];
+    __block BOOL hasBeenExplicitlyCanceled = false;
     // Request always a new token
     [self.logger verbose:@"RS:%p ARTAuthInternal [authorize.%@, delegate=%@]: requesting new token", _rest, authorizeId, lastDelegate ? @"YES" : @"NO"];
     NSObject<ARTCancellable> *task;
     self->_authorizationsCount += 1;
     task = [self _requestToken:currentTokenParams withOptions:replacedOptions callback:^(ARTTokenDetails *tokenDetails, NSError *error) {
         self->_authorizationsCount -= 1;
 
-        void (^successBlock)(void) = ^{
+        void (^successCallbackBlock)(void) = ^{
```

Comment:
The block pointers should be `const` (they're never explicitly set to `nil` anywhere, as they just fall out of scope, so we should make that clear).

Reply:
changed here https://github.com/ably/ably-cocoa/pull/1118/commits/23b90b26f1002fe137b9013d0a7b7e577e1dc8ca